collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java

```diff
@@ -550,9 +552,9 @@ private boolean processChunk(int low, int high, Outcome outcome, Long... values)
                     String id = c.getString(c.getColumnIndex(InstanceColumns._ID));
                     Uri toUpdate = Uri.withAppendedPath(InstanceColumns.CONTENT_URI, id);
 
-                    // Use the app's configured URL unless the form included a submission URL
+                    // Use the provided submission url OR the submission URL included in the form OR the app's configured URL in that order of priority/precedence
```

Comment:
This would be easier to read as a bulleted list, I think. Otherwise you can just wrap at around 80 chars.